src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -118,7 +196,9 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                     .flat_map(|imp| cx.tcx.associated_items(*imp).in_definition_order())
                     .any(|item| item.ident.name == variant_name)
                 {
-                    return Err(ErrorKind::ResolutionFailure);
+                    // This is just to let `fold_item` know that this shouldn't be considered;
+                    // it's a bug for the error to make it to the user
```

Comment:
```suggestion
                    // This was already found in another namespace. No need to return it again.
                    // This error is just to let `fold_item` know that this shouldn't be considered;
                    // it's a bug for the error to make it to the user
```
Hmm, on reflection it seems like there is some ambiguity here. Consider ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=24e1246fd29017cc498b98a6a2c80841)):
```rust
#[derive(Debug)]
#[allow(non_camel_case_types)]
pub enum E { f, g }
impl E { pub fn f() {} }

fn main() {
    println!("{:?}", E::f);
    println!("{:?}", E::f()); // error[E0618]: expected function, found enum variant `E::f`
    let f: fn() = E::f;  // error[E0308]: mismatched types: expected fn pointer, found enum `E`
    println!("{:?}", f);
}
```
`E::f` (the variant) and `E::f()` (the function) are both in the same namespace. It looks like the way `rustc` treats this is to just never let you access `E::f()`?? This is weird.

Reply:
To be consistent with rustc this should probably be switched in rustdoc so that `E::f` (the variant) has precedence. But this seems like a really weird part of the language.

Reply:
I think fixing this should go in a follow-up PR, it's a behavior change and not really related to diagnostics.